x-pack/plugins/endpoint/public/sagas/home.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AppMountContext } from 'kibana/public';
+import { withPageNavigationStatus } from './common';
+import { hrefIsForPath } from '../concerns/routing';
+import { actions as homeActions } from '../actions/home';
+import { SagaContext } from '../lib/saga';
+
+// TODO: type this properly
+export async function homeSaga(sagaContext: SagaContext, context: AppMountContext) {
+  await Promise.all([resourceSaga(sagaContext, context)]);
+}
+
+// TODO type actionsAndState, dispatch
+async function resourceSaga({ actionsAndState, dispatch }: SagaContext, context: AppMountContext) {
+  function isOnPage(href: any) {
+    return hrefIsForPath(href, `${context.core.http.basePath.get()}/app/endpoint/`);
+  }
+
+  for await (const {
+    action,
+    userIsOnPageAndLoggedIn,
+    href,
+    state,
+    shouldInitialize,
+  } of withPageNavigationStatus({
+    actionsAndState,
+    isOnPage,
+  })) {
+    if (userIsOnPageAndLoggedIn && shouldInitialize) {
+      try {
+        const data = await context.core.http.post('/endpoint/bootstrap', {});
+        dispatch(homeActions.serverReturnedBootstrapData(data));
```

Comment:
Just noticing that this is no longer user initialized, but rather done automatically when a user lands on `.../app/endpoint/` . Are you sure this is running only once? (example: play with the browser back|fwd buttons and place a breakpoint here on line 36)

Sorry I missed this before